tutorials/asr/ASR_with_NeMo.ipynb

```diff
@@ -0,0 +1,920 @@
+        "quartznet.change_vocabulary(new_vocabulary=[' ', 'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z', \"'\", \"!\"])"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "M7lvmiMSd3Aw",
+        "colab_type": "text"
+      },
+      "source": [
+        "After this, our decoder has completely changed, but our encoder (where most of the weights are) remained intact. Let's fine tune-this model for 2 epochs on AN4 dataset. We will also use the smaller learning rate from ``new_opt``."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "_PZJIso-eDl-",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "source": [
+        "# To use smaller learning rate\n",
+        "quartznet.setup_optimization(optim_config=DictConfig(new_opt))\n",
+        "# To point to the data we'll use for finetuning\n",
+        "quartznet.setup_training_data(train_data_config=params['model']['train_ds'])\n",
+        "# To point to the data we'll use for finetuning\n",
+        "quartznet.setup_training_data(train_data_config=params['model']['validation_ds'])\n",
+        "trainer = pl.Trainer(gpus=[1], max_epochs=2)\n",
+        "# TODO: OLEKSII FIX NEMO bug\n",
+        "# trainer.fit(quartznet)"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "RBcJtg5ulR0H",
+        "colab_type": "text"
+      },
+      "source": [
+        "## Model Improvements\n",
+        "\n",
+        "You already have all you need to create your own ASR model in NeMo, but there are a few more tricks that you can employ if you so desire. In this section, we'll briefly cover a few possibilities for improving an ASR model.\n",
+        "\n",
+        "### Data Augmentation\n",
+        "\n",
+        "There exist several ASR data augmentation methods that can increase the size of our training set.\n",
+        "\n",
+        "For example, we can perform augmentation on the spectrograms by zeroing out specific frequency segments (\"frequency masking\") or time segments (\"time masking\") as described by [SpecAugment](https://arxiv.org/abs/1904.08779), or zero out rectangles on the spectrogram as in [Cutout](https://arxiv.org/pdf/1708.04552.pdf). In NeMo, we can do all three of these by simply adding in a `SpectrogramAugmentation` neural module. (As of now, it does not perform the time warping from the SpecAugment paper.)\n",
+        "\n",
+        "Our toy model does not do spectrogram augmentation. But the real one we got from cloud does:"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "9glGogaPlR0H",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "source": [
+        "print(quartznet._cfg['spec_augment'])"
```

Comment:
Should probably also explain how one would have to add this to the config file to add this into the model.

Reply:
added